docs/io/registry.rst

```diff
@@ -24,80 +24,99 @@ for the :class:`~astropy.table.Table` and
+the ``read`` method.
+
+In the above case, we can write a simplistic function that only looks at
 filenames (but in practice, this function could even look at the first few
-bytes of the file for example). The only requirement is that it return a
-boolean indicating whether the input matches that expected for the format::
+bytes of the file for example). The only requirement for the identifier
+function is that it return a boolean indicating whether the input matches that
+expected for the format. In our example, we want to automatically recognize
+files with filenames ending in ``.mtf`` as being in the ``my-table-format``
+format::
 
-    def fits_identify(origin, *args, **kwargs):
-        return isinstance(args[0], basestring) and \
-               args[0].lower().split('.')[-1] in ['fits', 'fit']
+    from astropy.extern import six
+
+    def identify_mtf(origin, *args, **kwargs):
+        return (isinstance(args[0], six.string_types) and
+                args[0].lower().split('.')[-1] in ['mtf'])
 
 .. note:: Identifier functions should be prepared for arbitrary input - in
           particular, the first argument may not be a filename or file
           object, so it should not assume that this is the case.
 
-We then register this identifier function::
+We then register this identifier function, similarly to the reader function::
 
-    registry.register_identifier('fits', Table, fits_identify)
+    registry.register_identifier('my-table-format', Table, identify_mtf)
 
-And we can then do::
+Having registered this function, we can then do::
 
-    t = Table.read('catalog.fits')
+    t = Table.read('catalog.mtf')
 
 If multiple formats match the current input, then an exception is
 raised, and similarly if no format matches the current input. In that
 case, the format should be explicitly given with the ``format=``
 keyword argument.
 
-Similarly, it is possible to create custom writers. To go with our simplistic FITS reader above, we can write a simplistic FITS writer::
+It is also possible to create custom writers. To go with our custom reader
+above, we can write a custom writer::
+
+   def my_table_writer(table, filename, clobber=False):
+       ...  # Write the table out to a file
 
-   def fits_table_writer(table, filename, clobber=False):
-       import numpy as np
-       from astropy.io import fits
-       fits.writeto(filename, np.array(table), clobber=clobber)
+Writer functons should take a dataset object (either an instance of the
+:class:`~astropy.table.Table` or :class:`~astropy.nddata.NDData`
+classes or sub-classes), and any number of subsequent positional and keyword
+arguments - although as for the reader, the ``format`` keyword argument cannot
+be used.
 
 We then register the writer::
 
-   io_registry.register_writer('fits', Table, fits_table_writer)
+   registry.register_writer('my-custom-format', Table, my_table_writer)
 
-And we can then write the file out to a FITS file::
+and we can write the table out to a file::
```

Comment:
I'm not keen on sentences split by a code block.  This could be `We can then write the table out to a file::`.